Rename data columns

MaxBlesch opened the pull request:
@amageh and @MaxBlesch will rename the functions and Python variables accord to GEP-1. This will affect all modules. No other code should be changed while we do that.

hmgaudecker commented:
Please do not touch functions too much yet, there are a lot of constructs like this made-up example:

```arbeitsl_geld = ue_benfits(...)```

which would become

```arbeitsl_geld = arbeitsl_geld(...)```

under the rules of GEP-1.

This will only make sense once we implement the currently favoured backend proposal.

hmgaudecker commented:
The really important bit for the moment are the data columns.

MaxBlesch commented:
Okay. We just saw that. We will not change the functions too much and concentrate on the columns.

MaxBlesch commented:
After a lot of discussion and considerations, we suggest the following renaming: This is now to a limit of 16 characters. We couldn't do better!

| Current         | Proposal          |
|-----------------|-------------------|
| hid             | hid               |
| tu_id           | tu_id             |
| pid             | pid               |
| head_tu         | hh_vorstand       |
| adult_num       | anz_erw_hh        |
| child0_18_num   | anz_minderj_hh    |
| hh_wealth       | hh_vermögen       |
| m_wage          | lohn_m            |
| age             | alter             |
| selfemployed    | selbstständig     |
| east            | ostdeutsch        |
| haskids         | hat_kinder        |
| m_self          | eink_selbstst_m   |
| m_pensions      | rente_m           |
| pkv             | prv_krank_vers    |
| priv_pens_contr | prv_rente_beit_m  |
| m_wage_l1       | dur_eink_vorj_m   |
| months_ue       | m_arbeitsl        |
| months_ue_l1    | m_arbeitsl_vorj   |
| months_ue_l2    | m_arbeitsl_vor2j  |
| w_hours         | arbeitsstund_w    |
| child_num_tu    | anz_kinder_tu     |
| adult_num_tu    | anz_erw_tu        |
| byear           | geburtsjahr       |
| exper           | j_arbeitserf      |
| child           | kind              |
| pensioner       | renter            |
| m_childcare     | kind_betr_kost_m  |
| m_imputedrent   | eigenheim_ersp_m  |
| m_kapinc        | kapital_eink_m    |
| m_vermiet       | vermiet_eink_m    |
| miete           | kaltmiete_m       |
| heizkost        | heizkost_m        |
| renteneintritt  | rente_eint_alter  |
| handcap_degree  | behinderungsgrad  |
| wohnfl          | wohnfläche        |
| zveranl         | gem_veranlagt     |
| ineducation     | in_ausbildung     |
| alleinerz       | alleinerziehend   |
| eigentum        | wohneigentum      |
| cnstyr          | wohn_baujahr      |
| m_transfers     | sonstig_eink_m    |

Output columns:
| Current         | Proposal          |
|-----------------|-------------------|
| svbeit          | sozialv_beit_m    |
| rvbeit          | rentenv_beit_m    |
| avbeit          | arbeitsl_beit_m   |
| gkvbeit         | krankv_beit_m     |
| m_alg1          | arbeitsl_geld_m   |
| pensions_sim    | rente_anspr_m     |
| gross_e1        | brutto_eink_1     |
| gross_e5        | brutto_eink_5     |
| gross_e6        | brutto_eink_6     |
| gross_e7        | brutto_eink_7     |
| abgst           | abgelt_st         |
| soli            | soli_st           |
| kindergeld      | kindergeld        |
| incometax       | eink_st           |
| incometax_tu    | eink_st           |
| uhv             | unterhalt_vors_m  |
| regelbedarf     | regelbedarf_m     |
| regelsatz       | regelsatz_m       |
| alg2_kdu        | kost_unterk_m     |
| kiz             | kinderzuschl_m    |
| wohngeld        | wohngeld_m        |
| m_alg2          | arbeitsl_geld_2_m |
| dpi_ind         | verfüg_eink_m     |
| gross           | brutto_eink       |

We suggest to remove any aggregated output columns. Don't really see the need to sum two incomes for the end user!

Eric-Sommer commented:
> adult_num 	anz_erw_hh

I do see some room for confusing "Erwachsene" and "Erwerbstätige".


> pensioner 	renter

`rentner`

> m_imputedrent 	eigenheim_ersp_m

`ersp`? These are fictive rents / unterstellte Miete

> renteneintritt 	rente_eint_alter

This should be `rente_eint_jahr`

> cnstyr 	wohn_baujahr

`baujahr` might be enough here

> We suggest to remove any aggregated output columns. Don't really see the need to sum two incomes for the end user!

Agree, but please maintain `dpi`. This is the central variable for any distributional analysis.

hmgaudecker commented:
Thanks for this great proposal and all of your work! The below is just quickly going through stuff. Please see it as a basis for further discussion, not what I think the end result has to look like. Remember one should use the same care for naming variables as one takes for naming one's first child... 

| Current         | Proposal          | Proposal HMG    | Comment HMG       |
|-----------------|-------------------|-----------------|-------------------|
| hid             | hid               | hh_id           | Using `_hh` for col extensions, so why not here? And: Underscores are pretty :-) |
| tu_id           | tu_id             |                 | Will be irrelevant soon |
| pid             | pid               | p_id            | Underscores are pretty :-) |
| head_tu         | tu_vorstand       |                 | Will be irrelevant soon |
| adult_num       | anz_erw_hh        |                 | Will be irrelevant soon |
| child0_18_num   | anz_minderj_hh    |                 | Will be irrelevant soon |
| hh_wealth       | hh_vermögen       | vermögen_hh     | We should be consistent with prefix / postfix where possible. |
| m_wage          | lohn_m            | bruttolohn_m    | Better explicit...  |
| age             | alter             |                 | Drop this, use `geburtsjahr` only (should become `geburtsdatum` eventually). Or will that break current tests? |
| selfemployed    | selbstständig     |                 |                   |
| east            | ostdeutsch        | wohnort_ost     | Better explicit... |
| haskids         | hat_kinder        |                 | Will be irrelevant soon |
| m_self          | eink_selbstst_m   |                 |                   | 
| m_pensions      | rente_m           | ges_rente_m     | We may have private pensions, too |
| pkv             | prv_krank_vers    | prv_krankv_beit_m | 17 char should be fine, be consistent with public stuff in outputs |
| priv_pens_contr | prv_rente_beit_m  |                  |                   |
| m_wage_l1       | dur_eink_vorj_m   | bruttolohn_vorj_m | 17 characters should be fine |
| months_ue       | m_arbeitsl        | m_arbeitsl_lfdj  | Only relevant in conjunction with the two below, no? Then better explicit... |
| months_ue_l1    | m_arbeitsl_vorj   |                  |                   |
| months_ue_l2    | m_arbeitsl_vor2j  |                  |                   |
| w_hours         | arbeitsstund_w    |                  |                   |
| child_num_tu    | anz_kinder_tu     |                  | Will be irrelevant soon |
| adult_num_tu    | anz_erw_tu        |                  | Will be irrelevant soon |
| byear           | geburtsjahr       |                  | Change to geburtsdatum eventually. |
| exper           | j_arbeitserf      | arbeitserf_j     | Why relevant? If only for public pensions, we should be more explicit (ges_rente_beitr_j ?) |
| child           | kind              |                  | Will be irrelevant soon |
| pensioner       | rentner           |                  | We might have to be more precise what we mean. Somebody who claims pension benefits? Or somebody who does not work for pay? The two might not be the same... | 
| m_childcare     | kind_betr_kost_m  | betreuungskost_m | Will store that with the child eventually, so the `kind` bit should be obvious. But happy to leave as is until we make that change.
| m_imputedrent   | eigenheim_ersp_m  | ?                | Too close to "Ersparnis für Eigenheim" IMHO. What is the term in the law? Sorry time budget too tight to check right now. |
| m_kapinc        | kapital_eink_m    |                  | Why monthly?      |
| m_vermiet       | vermiet_eink_m    |                  | Why monthly?      |
| miete           | kaltmiete_m       |                  |                   |
| heizkost        | heizkost_m        |                  |                   |
| renteneintritt  | rente_eint_jahr   | jahr_renteneintr | We might have to be more precise what we mean. Somebody who claims pension benefits? Or somebody who does not work for pay? The two might not be the same... | 
| handcap_degree  | behinderungsgrad  |                  |                   |
| wohnfl          | wohnfläche        |                  |                   |
| zveranl         | gem_veranlagt     |                  | Will be irrelevant soon (#150)|
| ineducation     | in_ausbildung     |                  |                   |
| alleinerz       | alleinerziehend   |                  | Will be irrelevant soon |
| eigentum        | wohneigentum      |                  | What is meant by this? Whether someone owns house/flat? Its value? ... ? |
| cnstyr          | baujahr           | immobilie_baujahr | ?                |
| m_transfers     | sonstig_eink_m    | prv_transfers_m   | ?                |

Output columns:
| Current         | Proposal          | Proposal HMG    | Comment HMG       |
|-----------------|-------------------|-----------------|-------------------|
| svbeit          | sozialv_beit_m    |                 | Not just the sum of the four below? Then drop for the reasons you gave below |
| rvbeit          | rentenv_beit_m    |                  |                   |
| avbeit          | arbeitsl_beit_m   | arbeitsl_v_beit_m |                   |
| gkvbeit         | krankv_beit_m     | ges_krankv_beit_m | I guess it is okay to deviate from the "abbreviation implies underscore" rule here. consitency is the hobgloblin...|
| pvbeit          | pflegev_beit_m    |                  |                   |
| m_alg1          | arbeitsl_geld_m   |                  |                   |
| pensions_sim    | rente_anspr_m     |                  | What is this precisely?                  |
| gross_e1        | brutto_eink_1     |                  | ?                 |
| gross_e5        | brutto_eink_5     |                  | ?                 |
| gross_e6        | brutto_eink_6     |                  | ?                 |
| gross_e7        | brutto_eink_7     |                  | ?                 |
| abgst           | abgelt_st         |                  |                   |
| soli            | soli_st           |                  |                   |
| kindergeld      | kindergeld        |                  |                   |
| incometax       | eink_st           |                  | Is this even defined? I would tend to drop it |
| incometax_tu    | eink_st_tu        |                  |                   |
| uhv             | unterhalt_vors_m  | unterhaltsvors_m |                   |
| regelbedarf     | regelbedarf_m     |                  | Does it make sense to have these three?                  |
| regelsatz       | regelsatz_m       |                  | Does it make sense to have these three?                  |
| alg2_kdu        | kost_unterk_m     |                  | Does it make sense to have these three?                  |
| kiz             | kinderzuschlag_m  |                  |                   |
| wohngeld        | wohngeld_m        |                  |                   |
| m_alg2          | arbeitsl_geld_2_m |                  |                   |
| dpi_ind         | verfüg_eink_m     |                  |                   |
| gross           | brutto_eink       |                  | What is this precisely? Why relevant? If anything, I would have expected `verfügbares_eink`... |

MaxBlesch commented:
So, the code is working again. There is still a lot todo. For example, not all internal columns are yet marked with an underscore in the beginning. @amageh and I will probably meet next week in person to implement the remaining things together. Til then I vote for a merge, that others can contribute. Maybe we can also meet in person @hmgaudecker to discuss your proposals. At least I  have some questions, that could be better discussed in person. If nobody else has some major objections, maybe @Eric-Sommer or @hmgaudecker could approve!

hmgaudecker commented:
Sounds great, thanks.

I do not think we need to touch that underscores upfront thing here, i should mal that as tentative in the GEP. If the backend works out as envisaged, we will not need it.

hmgaudecker commented on gettsim/benefits/arbeitsl_geld.py:
```diff
@@ -14,7 +14,7 @@ def ui(
     """
     # Beitragsbemessungsgrenze differs in east and west germany
-    westost = "o" if person["ostdeutsch"] else "w"
+    westost = "o" if person["wohnort_st"] else "w"
```
I think it should be "`wohnort_ost`"

amageh replied:
You are correct, thank you !

amageh commented:
In order to keep an overview of the variable names, I will post the most current version here. This list contains all the variables we have changed since the beginning. The lists in the documentation will be reduced to the set of variables we have discussed. This is just to stay on top of the changes we make.

Inputs:
| Before            | Now               | Comment                  |
|-------------------|-------------------|--------------------------|
| hid               | hh_id             |                          |                         
| tu_id             | tu_id             | To be removed            | 
| pid               | p_id              |                          |
| tu_vorstand       | tu_vorstand       | To be removed            | 
| anz_erw_hh        | anz_erwachsene_hh | To be removed            | 
| anz_minderj_hh    | anz_minderj_hh    | To be removed            |
| hh_vermögen       | vermögen_hh       |                          |
| lohn_m            | bruttolohn_m      |                          |
| alter             | alter             | Drop, only keep `geburtsdatum` |
| selbstständig     | selbstständig     |                          |       
| ostdeutsch        | wohnort_ost       |                          |
| hat_kinder        | hat_kinder        | To be removed            |
| eink_selbstst_m   | eink_selbstst_m   |                          |        
| rente_m           | ges_rente_m       |                          | 
| prv_krank_vers    | prv_krankv_beit_m |                          | 
| prv_rente_beit_m  | prv_rente_beit_m  |                          |                    
| dur_eink_vorj_m   | bruttolohn_vorj_m |                          | 
| m_arbeitsl        | arbeitsl_lfdj_m   |                          |
| m_arbeitsl_vorj   | arbeitsl_vorj_m   |                          |
| m_arbeitsl_vor2j  | arbeitsl_vor2j_m  |                          |
| arbeitsstund_w    | arbeitsstunden_w  |                          |
| anz_kinder_tu     | anz_kinder_tu     | To be removed            |
| anz_erw_tu        | anz_erw_tu        | To be removed            |
| geburtsjahr       | geburtsjahr       | Change to `geburtsdatum`    |
| j_arbeitserf      | arbeitserf_j      |                          |
| kind              | kind              | To be removed            |
| rentner           | rentner           |                          |
| kind_betr_kost_m  | betreuungskost_m  |                          |
| eigenheim_ersp_m  | miete_unterstellt |                          | 
| kapital_eink_m    | kapital_eink_m    |                          |       
| vermiet_eink_m    | vermiet_eink_m    |                          |       
| kaltmiete_m       | kaltmiete_m       |                          |                   
| heizkost_m        | heizkost_m        |                          |                   
| rente_eint_jahr   | jahr_renteneintr  | Change to `datum_renteneintritt` | 
| behinderungsgrad  | behinderungsgrad  |                          |              
| wohnfläche        | wohnfläche        |                          |              
| gem_veranlagt     | gem_veranlagt     | To be removed            | 
| in_ausbildung     | in_ausbildung     |                          |              
| alleinerziehend   | alleinerziehend   | To be removed            | 
| wohneigentum      | bewohnt_eigentum  |                          |              
| baujahr           | immobilie_baujahr |                          |            
| sonstig_eink_m    | prv_transfers_m   | To be removed            |            

Outputs:
| Before            | Now               | Comment           |
|-------------------|-------------------|-------------------|
| sozialv_beit_m    | sozialv_beit_m    |                   |
| rentenv_beit_m    | rentenv_beit_m    |                   |
| arbeitsl_beit_m   | arbeitsl_v_beit_m |                   |
| krankv_beit_m     | ges_krankv_beit_m |                   |
| pflegev_beit_m    | pflegev_beit_m    |                   |
| arbeitsl_geld_m   | arbeitsl_geld_m   |                   |
| rente_anspr_m     | rente_anspr_m     |                   |
| brutto_eink_1     | brutto_eink_1     | To be removed |
| brutto_eink_5     | brutto_eink_5     | To be removed |
| brutto_eink_6     | brutto_eink_6     | To be removed |
| brutto_eink_7     | brutto_eink_7     | To be removed |
| abgelt_st         | abgelt_st         |                   |
| soli_st           | soli_st           |                   |
| kindergeld        | kindergeld        |                   |
| eink_st           | eink_st           |                   |
| eink_st_tu        | eink_t_tu         |                   |
| unterhalt_vors_m  | unterhaltsvors_m  |                   |
| regelbedarf_m     | regelbedarf_m     | To be removed     |
| regelsatz_m       | regelsatz_m       |                   |
| kost_unterk_m     | kost_unterk_m     |                   |
| kinderzuschlag_m  | kinderzuschlag_m  |                   |
| wohngeld_m        | wohngeld_m        |                   |
| arbeitsl_geld_2_m | arbeitsl_geld_2_m |                   |
| verfüg_eink_m     | verfüg_eink_m     |                   |
| dpi               | verfüg_eink_hh_m  |					|
| brutto_eink       | brutto_eink       | To be removed     |

Eric-Sommer commented:
Very good. Two little remarks:
- `hat_kinder` might refer to children outside the household (matters for Pflegeversicherungsbeitrag) and can hence not be interferred from the household composition.
- why drop `brutto_eink`? The difference between distributions of gross and disposable income is of key interest for many distributional analyses.

MaxBlesch commented:
So before the name was `haskids`which could have implied the same. I think with the documentation this becomes clear.

We didn't find really a good name for `brutto_eink`, because we just add up all incomes. Do we really need a variable for this or could that not easily be done by the end-user as well?

hmgaudecker commented:
We will keep the kids-outside-the-household thing in mind when restructuring the dataset requirements.

We put off `brutto_eink` for the moment because the name is not precise enough. With labor income and taxes only, it is obvious enough. With transfers, much less so. And how to deal with Werbungskosten and similar tax deductions? E.g., if I have annual income from renting out a flat =20k and pay for 10k interest and other cost -- is by `brutto_eink` 20k or 10k?

So I think it is clear that we will add something like this back to the interface in the not-so-distant future, but it needs to be more precise as GETTSIM allows for use cases not tied quite as closely to what is recorded well in the SOEP.

amageh commented:
This concludes our first draft for the variable renaming. @hmgaudecker & @Eric-Sommer, please let us know if there are any other changes you would like to see implemented here. Otherwise @MaxBlesch & I will merge our changes tomorrow.

hmgaudecker commented on docs/getting-started.rst:
```diff
@@ -161,74 +158,45 @@ Columns returned by the simulator
++-------------------------+----------------------------------------------------+-------+
+| soli_st                 | Monthly solidarity surcharge due, individual       | float |
++-------------------------+----------------------------------------------------+-------+
+| kindergeld              | Monthly child Benefit, individual                  | float |
```
Make this `kindergeld_m` (I would think of the general rule for now: Assume annually unless specified otherwise).

amageh replied:
Has been changed to `kindergeld_m`.

hmgaudecker commented on docs/getting-started.rst:
```diff
@@ -161,74 +158,45 @@ Columns returned by the simulator
++-------------------------+----------------------------------------------------+-------+
+| rente_anspr_m           | Monthly amount of old-age pensions                 | float |
++-------------------------+----------------------------------------------------+-------+
+| abgelt_st               | Monthly capital income tax due, individual         | float |
```
All taxes should be annual, I believe:
- `abgelt_st`
- `soli_st`
- `eink_st`
- `eink_st_tu`

Taxes are just filed once per year, so it does not make much sense to divide by 12. 

Eventually, we probably want to allow for much richer ways of doing things, but that is for a much later date.

Eric-Sommer replied:
I'd argue for keeping them in monthly terms in the output at least for now, because it is very easy to mix up quantities.

hmgaudecker replied:
At a minimum, we should append `_m` to the names then. No super-strong opinion here.

amageh replied:
Good point. I have added the `_m` to all of the taxes.

hmgaudecker commented on docs/getting-started.rst:
```diff
@@ -161,74 +158,45 @@ Columns returned by the simulator
++-------------------------+----------------------------------------------------+-------+
+| arbeitsl_geld_2_m       | Monthly social assistance, household sum           | float |
++-------------------------+----------------------------------------------------+-------+
+| verfüg_eink_m           | Monthly disposable income, individual              | float |
```
I would prefer to add a "b": `verfügb_eink`. Thoughts?

amageh replied:
Changed it!

hmgaudecker commented on gettsim/benefits/arbeitsl_geld.py:
```diff
@@ -49,13 +49,13 @@ def proxy_net_wage_last_year(
 
     # Fictive taxes (Lohnsteuer) are approximated by applying the wage to the tax tariff
-    prox_tax = e_st_params["tax_schedule"](12 * max_wage - werbungs_pausch, e_st_params)
+    prox_tax = e_st_params["st_tarif"](12 * max_wage - werbungs_pausch, e_st_params)
```
Please change `e_st` to `eink_st` everyhwere, including the yaml file name etc. (Only leave ESt as the abbreviation of the law)

MaxBlesch replied:
Changed to `eink_st` everywhere!

hmgaudecker commented on gettsim/benefits/arbeitsl_geld_2.py:
```diff
@@ -12,30 +12,39 @@ def alg2(household, params):
-    household["ar_alg2_ek"] = np.maximum(
-        household["alg2_ek"] - household["ekanrefrei"], 0
+    household["sum_arbeitsl_geld_2_eink"] = np.maximum(
+        household["arbeitsl_geld_2_eink"] - household["eink_anrech_frei"], 0
```
I would prefer an additional "n": `eink_anrechn_frei`. Thoughts?

amageh replied:
Done.

hmgaudecker commented on gettsim/benefits/elterngeld.py:
```diff
@@ -172,47 +175,47 @@ def check_eligibilities(household, params):
     household["elternzeit_anspruch"] = False
     household["geschw_bonus"] = False
-    household["num_mehrlinge"] = 0
+    household["anz_mehrlinge"] = 0
```
Can we just use something that counts kids born on one date rather than subtracting 1? I find this very confusing and it is impossible for a non-German speaker to find out what we mean.

MaxBlesch replied:
It is really hard for me to think of the right variable. I mean, if a non-english speaker translates mehrlinge, he/she will exactly get multiples and that's what we count. I would add a comment, which clarifies the notion?

hmgaudecker replied:
But when thinking of multiples, twins would get a value of 2, triplets of 3, ... IIUC, this would count twins as anz_mehrlinge=1 ?

MaxBlesch replied:
True. But if I change the counting, a single child gets anz_mehrlinge=1 and it is not a mehrling. I found that confusing as well. I now changed to a bonus variable. Maybe that works!

hmgaudecker commented on docs/getting-started.rst:
```diff
@@ -58,100 +58,97 @@ long format, i.e. each person no matter what age has his/her own row.
+| immobilie_baujahr   | Construction year of dwelling                          | int   |
+|                     | (1: <1965,2:1966-2000,3:>2000)                         |       |
++---------------------+--------------------------------------------------------+-------+
+| prv_transfers_m     | Sum of monthly public/private transfers not simulated. | int   |
```
Add a note that this will be split up soon.

amageh replied:
I made a small addition to the explanation in the documentation for this.

hmgaudecker commented on gettsim/social_insurance.py:
```diff
@@ -1,4 +1,10 @@
-OUT_COLS = ["svbeit", "rvbeit", "avbeit", "gkvbeit", "pvbeit"]
+OUT_COLS = [
+    "sozialv_beit_m",
```
This first can go, right?

MaxBlesch replied:
We keep it as an internal variable and do not return it to the user. We use it in several calculations and it makes life just easier! Is this a real problem?

hmgaudecker replied:
Nope!

hmgaudecker commented on gettsim/tax_transfer.py:
```diff
@@ -132,45 +137,47 @@ def tax_transfer(
+        "alleinerziehend",
+        "alter",
+        "anz_kinder_tu",
         "year",
```
I think we should go for `jahr`, shouldn't we?

MaxBlesch replied:
Changed everywhere to `jahr`. The python variables are left with `year`. We tackle functions and variables separatly, once we set up the new interfaces!

hmgaudecker commented on gettsim/policy_for_date.py:
```diff
@@ -3,27 +3,27 @@
+from gettsim.taxes.soli_st import keine_soli_st
+from gettsim.taxes.soli_st import soli_st_formel_1991_92
+from gettsim.taxes.soli_st import soli_st_formel_seit_1995
 from gettsim.taxes.zve import calc_hhfreib_from2015
```
Please add an underscore according to the variable that is calculated (`hh_freib`). Also make such changes elsewhere. Makes it much easier to search & replace later on.

MaxBlesch replied:
I didn't want to touch functions yet. I wanted to postpone when setting up the new interface!

hmgaudecker replied:
[A Foolish Consistency is the Hobgoblin of Little Minds](A Foolish Consistency is the Hobgoblin of Little Minds)

hmgaudecker commented on docs/getting-started.rst:
```diff
@@ -161,74 +158,45 @@ Columns returned by the simulator
++-------------------------+----------------------------------------------------+-------+
+| verfüg_eink_hh_m        | Monthly disposable income including household      | float |
+|                         | level benefits, household sum                      |       |
++-------------------------+----------------------------------------------------+-------+
```
Don't we return `elterngeld` now? Should probably be `elterngeld_m`, too, by the argument above.

Please go through [all changes since 0.2.1](https://gettsim--156.org.readthedocs.build/en/156/changes.html#tbd) and check whether there are other cases where inputs/outputs require adjustment.

MaxBlesch replied:
Included `elterngeld_m`. I went through the changes. I think we caught all. No claim of completness though.

hmgaudecker commented:
I would also be good, I think, to add the cross-walk of columns in 0.2 and columns in 0.3 to the documentation. This can go probably go again in the 0.4 docs, but it will be a valuable resource to those familiar with the IZAΨMOD codebase.

MaxBlesch commented:
> It would also be good, I think, to add the cross-walk of columns in 0.2 and columns in 0.3 to the documentation. This can probably go again in the 0.4 docs, but it will be a valuable resource to those familiar with the IZAΨMOD codebase.

I would create an issue for that or just open a new PR. But if there are no other concerns I would really like to merge this PR! Then we have a new base to start the regrouping!

hmgaudecker commented:
> > It would also be good, I think, to add the cross-walk of columns in 0.2 and columns in 0.3 to the documentation. This can probably go again in the 0.4 docs, but it will be a valuable resource to those familiar with the IZAΨMOD codebase.
> 
> I would create an issue for that or just open a new PR. But if there are no other concerns I would really like to merge this PR! Then we have a new base to start the regrouping!

What can be more relaxing after a long day than a glass of red wine or some table formatting? :laughing: ? 

Please double-check 6e8aa65.

hmgaudecker commented on CHANGES.rst:
```diff
@@ -22,13 +22,14 @@ releases are available on `Anaconda.org <https://anaconda.org/gettsim/gettsim>`_
   parameters and add Soli tests (:ghuser:`mjbloemer`, :ghuser:`MaxBlesch`)
 * Implement pre-2010 ruling on *Vorsorgeaufwendungen* (:ghuser:`Eric-Sommer`)
 * ``gettsim`` is released as a conda noarch package (:ghuser:`tobiasraabe`)
+* Renamed all data columns to german names (:ghuser:`amageh`, :ghuser:`MaxBlesch` )
```
Nit-picky: Please use correct capitalisation (German); I updated it in the table with columns already

MaxBlesch replied:
It is German now!

hmgaudecker commented:
Great work!

MaxBlesch commented:
So, I found another little thing. We required former `exper` as input, that wasn't necessary. The variable for earning points was `EP`, which is now `entgelt_punkte`. This variable gets updated and I added it to the returns. Also, I added it everywhere in the documentation.

hmgaudecker commented on docs/crosswalk.rst:
```diff
@@ -66,7 +66,7 @@ Required input columns
 | byear           | geburtsjahr       | Year of birth                                  |
 +-----------------+-------------------+------------------------------------------------+
-| exper           | j_arbeitserf      | Earning points for pension claim               |
+| EP              | entgelt_punkte    | Earning points for pension claim               |
```
entgeltpunkte should be one word.

MaxBlesch replied:
Changed to `entgeldpunkte`.

hmgaudecker replied:
I hope not :-) "t" not "d"

MaxBlesch replied:
Oh yeah. Changed to `entgeltpunkte`
